docs/src/lib/types.md

```diff
@@ -57,6 +57,53 @@ The `RepeatedVector` and `StackedVector` types are subtypes of `AbstractVector`
 with the exception that they are read only. Note that they are not exported and should not be constructed directly,
 but they are columns of a `DataFrame` returned by `stackdf` and `meltdf`.
 
+## [The design of handling of columns of a `DataFrame`](@id man-columnhandling)
+
+When a `DataFrame` is constructed columns are copied by default. You can disable
+this behavior by setting `copycols` keyword argument to `false`. The exception is
+if an `AbstractRange` is passed as a column, then it is always collected to a `Vector`.
+
+Also functions that transform a `DataFrame` to produce a new `DataFrame` perform a copy of the columns,
+unless they are passed `copycols=false` (available only for functions
+that could perform a transformation without copying the columns). Examples of such functions are [`vcat`](@ref),
+[`hcat`](@ref), [`filter`](@ref), [`dropmissing`](@ref), [`join`](@ref), `getindex`,
+[`copy`](@ref) or the [`DataFrame`](@ref) constructor mentioned above.
+
+On the contrary, functions that create a view of a `DataFrame` *do not* by definition make copies of
+the columns, and therefore require particular caution. This includes `view`, which returns
+a `SubDataFrame` or a `DataFrameRow`, and `groupby`, which returns a `GroupedDataFrame`.
+
+A partial exception to this rule are the [`stackdf`](@ref) and [`meltdf`](@ref) functions which
+create a `DataFrame` that contains views of the columns from the source `DataFrame`.
+
+In-place functions whose names end with `!` (like `sort!` or [`dropmissing!`](@ref),
+`setindex!`, `push!`, `append!`) may mutate the column vectors of the `DataFrame` they take
+as an argument. These functions are safe to call due to the rules described above,
+*except* when a view of the `DataFrame` is in use (via a `SubDataFrame`, a `DataFrameRow`
+or a `GroupedDataFrame`). In the latter case (i.e. when you call such a function on the parent of the view),
+the view might become corrupted, which make trigger errors, silently return invalid data or even cause Julia to crash.
+
+It is possible to have a direct access to a column `col` of a `DataFrame` `df`
+(e.g. this can be useful in performance critical code to avoid copying),
+using one of the following methods:
+
+* via the `getproperty` function using the syntax `df.col`;
+* via the `getindex` function using the syntax `df[:col]`; (note this is in contrast to `df[:, :col]` which copies)
+* by creating a `DataFrameColumns` object using the [`eachcol`](@ref) function;
+* by calling `parent` function on a view of a column of the `DataFrame`, e.g. `parent(@view(:, :col))`;
```

Comment:
```suggestion
* by calling `parent` function on a view of a column of the `DataFrame`, e.g. `parent(@view df[:, :col])`;
```

But beyond that perhaps we should say something about:

```
     * Note that you can just work with `@view df[:, :col]` rather than its `paraent` as it itself will be a noncopying complete view
```

Reply:
@nalimilan - do you think it is worth to add this additional comment (it is true, but it is unrelated and rather advanced topic, so someone who writes such a code should be aware of it so I am a bit hesitant in order to avoid complicating the documentation)?

Reply:
@oxinabox - thank you for the fix

Reply:
I agree it's not completely relevant here. That list is already long and complex enough.

Small suggestion:
```suggestion
* by calling the `parent` function on a view of a column of the `DataFrame`, e.g. `parent(@view(:, :col))`;
```

Reply:
suggestion added